templates/common/specutil/spec.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package spec contains commonly used function for handling spec files
+package specutil
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	spec "github.com/abcxyz/abc/templates/model/spec/v1beta1"
+)
+
+const (
+	// The spec file is always located in the template root dir and named spec.yaml.
+	SpacFileName = "spec.yaml"
+
+	// Keys for output formatting.
+	OutputDescriptionKey       = "Description"
+	OutputInputNameKey         = "Input name"
+	OutputInputDefaultValueKey = "Default"
+	OutputInputRuleKey         = "Rule"
+)
+
+// ParseSpecToList parses a spec.Spec into a list with
+// the keys and attribute pairs.
+//
+// Example:
+// ["Description", "example description", "Input Name", "example name"].
+func ParseSpecToList(spec *spec.Spec) []string {
+	l := make([]string, 0)
```

Comment:
This might be little more understandable if it was a [][]string instead of a []string. Then each line of text could be it's own list, rather than using the assumption that every index multiple of 2 starts a new line.

Reply:
Done.